enforce disk quotation limit only in public_html

bdgreenweb opened the pull request:
lock only public_html directory when user exceed disk usage limit. Openlitespeed keep restarting when it failed to write logs and cache file inside users home/domain.com/logs/ directory. moreover user cant remove files from trash folder after directory become immutable. It's better to lock only public_html directory upon exceeding the disk quota limit.

bdgreenweb committed:
enforce disk quotation limit only in public_html

lock only public_html directory when user exceed disk usage limit. Openlitespeed keep restarting when it failed to write logs and cache file inside users home/domain.com/logs/ directory. moreover user cant remove files from trash folder after directory become immutable. It's better to lock only public_html directory upon exceeding the disk quota limit.

usmannasir commented:
How about child domains? They can continue to write if they are out of the document root of master domain.

bdgreenweb committed:
Update IncScheduler.py

This will exclude log directory & backup directories from being immutable after exceeding disk quota limit.

bdgreenweb committed:
Update IncScheduler.py

bdgreenweb commented:
> How about child domains? They can continue to write if they are out of the document root of master domain.

I'm sorry @usmannasir Sir, I forgot about child domain and user created custom directory outside public_html. We can use static system in this case. First we will make entire directory immutable later we will unlock only .trash, logs, backup, incbackup directory, I've edited the file again. Please check now.

Thanks.

SaJeTek-Developer commented:
I also ran into this issue where all sites would not load because openlitespeed keeps crashing, this was because the lscache directory was in the vhost path also and that was causing the crashes. /home/domain.com/lscache. This path MUST also be considered when making the directory immutable.

bdgreenweb commented:
@SaJeTek-Developer 
Added lscache directory https://github.com/usmannasir/cyberpanel/pull/807/commits/6ca22b325f0fc0c16b2e9c87980dcd3c603d8573

SaJeTek-Developer commented:
Hi,

To add to this from my teething process. On a system running cloudlinux, cagefs also uses the /home/domain.com/.cagefs directory to write temporary files. Php runs into problems eventually and hangs when this directory is immutable.

SaJeTek-Developer commented:
Hi,

To add to this from my teething process. On a system running cloudlinux, cagefs also uses the /home/domain.com/.cagefs directory to write temporary files. Php runs into problems eventually and hangs when this directory is immutable.
